### DISCUSSION TITLE
An idea for `meson wrap init <your-project>` for generating a wrap file for a given project

### DISCUSSION THREAD

**Discussion by dreamer-coding** [mesonbuild/meson#13254] [Ideas]
What if we had Meson help generate the wrap files? Realistically it would know more information about a project like version number, what options are defined, and other attributes, and it would probably put an end to broken hash values using its sanity check system.

**Comment by eli-schwartz** [comment]
Wrap files don't describe project options to begin with.

Nor version numbers, actually, except inasmuch as the wrapdb associates a wrap file to the upstream project version it is tied to. Version numbers are *usually* part of the url but that isn't a requirement.

As far as hash values go, meson will print the hash value it actually got when a failed download happens, so you can just copy that. But it doesn't seem quite practical to generate a wrap file without knowing what url to use. So I don't know how you want to generate them.

It also doesn't answer the problem of defining the right `[provide]` section, which actually requires human judgement since not all declared dependencies are meant to be exposed, and conditional logic can result in meson not parsing a declared dependency definition in the first place.

**Reply by dreamer-coding** [reply]
Ok so no good argument for the provide section but as for the hash value perhaps a temporary directory downloads the archive and uses that for the hash value, when hash value is produced uses that. This would require that the release exist.

**Comment by dreamer-coding** [comment]
Could add sub-arguments for the [provide] section and other bits of info.

Something like:
```shell
meson wrap init <wrap name> --provide <project_dep_value>
meson wrap init <wrap name> --provide <project_dep_value> --version <xyz>
meson wrap init <wrap name> --provide <project_dep_value> --type <git/wrap/svc>
```
